yt_dlp/extractor/tiktok.py

```diff
@@ -793,102 +792,150 @@ def _real_extract(self, url):
+            'aid': '1988',
+            'app_language': 'en',
+            'app_name': 'tiktok_web',
+            'browser_language': 'en-US',
+            'browser_name': 'Mozilla',
+            'browser_online': 'true',
+            'browser_platform': 'Win32',
+            'browser_version': '5.0 (Windows)',
+            'channel': 'tiktok_web',
+            'cookie_enabled': 'true',
+            'count': '15',
+            'cursor': cursor,
+            'device_id': ''.join(random.choices(string.digits, k=19)),
+            'device_platform': 'web_pc',
+            'focus_state': 'true',
+            'from_page': 'user',
+            'history_len': '2',
+            'is_fullscreen': 'false',
+            'is_page_visible': 'true',
+            'language': 'en',
+            'os': 'windows',
+            'priority_region': '',
+            'referer': '',
+            'region': 'US',
+            'screen_height': '1080',
+            'screen_width': '1920',
+            'secUid': sec_uid,
+            'type': '1',  # pagination type: 0 == oldest-to-newest, 1 == newest-to-oldest
+            'tz_name': 'UTC',
+            'verifyFp': 'verify_%s' % ''.join(random.choices(string.hexdigits, k=7)),
+            'webcast_language': 'en',
         }
 
+    def _entries(self, sec_uid, user_name):
+        cursor = int(time.time() * 1E3)
         for page in itertools.count(1):
-            for retry in self.RetryManager():
-                try:
-                    post_list = self._call_api(
-                        'aweme/post', query, username, note=f'Downloading user video list page {page}',
-                        errnote='Unable to download user video list')
-                except ExtractorError as e:
-                    if isinstance(e.cause, json.JSONDecodeError) and e.cause.pos == 0:
-                        retry.error = e
-                        continue
-                    raise
-            yield from post_list.get('aweme_list', [])
-            if not post_list.get('has_more'):
+            response = self._download_json(
+                self._API_BASE_URL, user_name or sec_uid, f'Downloading page {page}',
+                query=self._build_web_query(sec_uid, cursor), headers={'User-Agent': self._USER_AGENT})
+
+            for video in traverse_obj(response, ('itemList', lambda _, v: v['id'])):
+                video_id = video['id']
+                webpage_url = self._create_url(user_name, video_id)
+                info = try_call(
+                    lambda: self._parse_aweme_video_web(video, webpage_url, video_id)) or {'id': video_id}
+                info.pop('formats', None)
+                yield self.url_result(webpage_url, TikTokIE, **info)
+
+            old_cursor = cursor
+            cursor = traverse_obj(
+                response, ('itemList', -1, 'createTime', {lambda x: x * 1E3}, {int_or_none}))
+            if not cursor:
+                cursor = old_cursor - 604800000  # jump 1 week back in time
```

Comment:
```suggestion
                cursor = old_cursor - 7 * 86_400_000  # jump 1 week back in time
```
is more readable imo. As for the comment, it may be more useful to explain "why 1 week"